docs/website/postprocess_dialect_docs.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# MLIR dialect markdown document postprocessor for website generation.
+#
+# Usage (typically invoked by generate_extra_files.sh):
+#   postprocess_dialect_docs.py ${WEBSITE_DOCS_BUILD_DIRECTORY}
+
+import argparse
+import os
+import fileinput
+import re
+
+
+def main(args):
+    directory = args.directory
+    files = [os.path.join(directory, f) for f in os.listdir(directory)]
+
+    # Replace certain headings depth 5, lining up with the table-of-contents
+    # (toc) max depth setting of 4 (`toc_depth: 4`).
```

Comment:
This sounds like you're saying you're replacing headings that are *already* depth 5. Also I think there's another reasonable justification, which is that it makes sense for all ops to have the same heading level regardless of whether or not they're in a group. Although you're actually not shifting heading-3 ops to be heading-4. If that's not too hard, I think I would do that too.

```suggestion
    # Move some headings from depth 4 to depth 5, lining up with the
    # table-of-contents (toc) max depth setting of 4 (`toc_depth: 4`).
```

More generally, could we consider upstreaming this as an option to the doc generator? I think it makes sense for ops to always be at the same heading level.

Reply:
Reworded.

> Although you're actually not shifting heading-3 ops to be heading-4. If that's not too hard, I think I would do that too.

I don't see how with existing dialects / tblgen we could see heading-3 text matching these patterns. If we did, I might want to reevaluate this approach entirely...

> More generally, could we consider upstreaming this as an option to the doc generator? I think it makes sense for ops to always be at the same heading level.

I'm not sure about upstreaming the heading depth changes, since skipping heading levels isn't quite standard / accepted everywhere. @jpienaar had some ideas at https://reviews.llvm.org/D153306#4438226

Reply:
> I don't see how with existing dialects / tblgen we could see heading-3 text matching these patterns. If we did, I might want to reevaluate this approach entirely...

I mean shift the op heading that's only depth 3 down to depth 4, so that it will be at the same depth as other ops that are in groups.

> > More generally, could we consider upstreaming this as an option to the doc generator? I think it makes sense for ops to always be at the same heading level.
> 
> I'm not sure about upstreaming the heading depth changes, since skipping heading levels isn't quite standard / accepted everywhere. @jpienaar had some ideas at https://reviews.llvm.org/D153306#4438226

Yeah makes sense. Creating an explicit "other" group SGTM

Reply:
Ah, my solution there is to ensure that every op is in a group (or don't use groups at all). That isn't quite working as intended today since some groups have only 1 op and those "groups" are dissolved (https://reviews.llvm.org/D153306#4442952)